setup.sh

```diff
@@ -0,0 +1,119 @@
+sudo apt update
+sudo apt install gcc-9 g++-9 python python3 ghc rvm imagemagick mysql-server libmysqlclient-dev libcurl4-openssl-dev libcap-dev openssl -y
+sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 60 \
+    --slave /usr/bin/g++ g++ /usr/bin/g++-9 \
+    --slave /usr/bin/gcc-nm gcc-nm /usr/bin/gcc-nm-9 \
+    --slave /usr/bin/gcc-ar gcc-ar /usr/bin/gcc-ar-9 \
+    --slave /usr/bin/gcc-ranlib gcc-ranlib /usr/bin/gcc-ranlib-9 # Use GCC 9 as default
+
+if [[ $ubuntu_distribution == "20.04" ]]; then
+    sudo mysql <<< "ALTER USER '$DB_USERNAME'@'localhost' IDENTIFIED WITH mysql_native_password BY '$DB_PASSWORD'; flush privileges;"
+fi
+sudo usermod -a -G rvm $USER
+
+echo 'source /etc/profile.d/rvm.sh' >> ~/.bashrc
+echo "PATH=$HOME/.rvm/gems/ruby-2.6.5/bin:$PATH" >> ~/.bashrc
+
+newgrp rvm << RVMGRPEOF
+
+set -x
+
+source /etc/profile.d/rvm.sh
+PATH=$HOME/.rvm/gems/ruby-2.6.5/bin:$PATH
+
+rvm install 2.6.5
+rvm alias create default 2.6.5
+
+RVMGRPEOF
+
+source /etc/profile.d/rvm.sh
+PATH=$HOME/.rvm/gems/ruby-2.6.5/bin:$PATH
+
+gem install rails -v 4.2.11
+
+git clone https://github.com/TIOJ-INFOR-Online-Judge/tioj $TIOJ_PATH
+git clone https://github.com/TIOJ-INFOR-Online-Judge/miku $MIKU_PATH
+cd $MIKU_PATH
+git submodule update --init
+
+cd $TIOJ_PATH
+bundle install
+
+echo "Please choose ruby as main programming language and use default settings in next interactive window."
+read -n 1 -p "Press any key to continue."
+ 
+rvmsudo_secure_path=1 /usr/share/rvm/bin/rvmsudo /usr/share/rvm/gems/ruby-2.6.5/bin/passenger-install-nginx-module
+
+sudo sed -i '/http {/a \ \ \ \ passenger_app_env production;' /opt/nginx/conf/nginx.conf
+sudo sed -i "/^[^#]*server_name/a \ \ \ \ \ \ \ \ passenger_enabled on;\n\ \ \ \ \ \ \ \ root $TIOJ_PATH/public;" /opt/nginx/conf/nginx.conf
+sudo sed -i "/^\ *location \/ {/, /}/ s|^\(\ \{8\}\)|\1# |" /opt/nginx/conf/nginx.conf
+
+cat <<EOF > config/database.yml
+default: &default
+  adapter: mysql2
+  encoding: utf8
+  pool: 5
+  username: $DB_USERNAME
+  password: $DB_PASSWORD
+  socket: /var/run/mysqld/mysqld.sock
+
+development:
+  <<: *default
+  database: tioj_development
+
+test:
+  <<: *default
+  database: tioj_test
+
+production:
+  <<: *default
+  database: tioj_production
+EOF
+
+TOKEN=$(rake secret)
+KEY=$(rake secret)
+
+sed "s/\".*\"/\"$TOKEN\"/" $TIOJ_PATH/config/initializers/secret_token.rb.example > $TIOJ_PATH/config/initializers/secret_token.rb
+sed "s/\".*\"/\"$KEY\"/" $TIOJ_PATH/config/initializers/fetch_key.rb.example > $TIOJ_PATH/config/initializers/fetch_key.rb
+
+sed "s,l.*,l = $URL,; s/y.*/y = $KEY/" $MIKU_PATH/app/tioj_url.py.example > $MIKU_PATH/app/tioj_url.py
+
+cd $TIOJ_PATH
+RAILS_ENV=production rake db:create db:schema:load db:seed
+RAILS_ENV=production rake assets:precompile
+mkdir public/announcement
+echo -n '{"name":"","message":""}' > public/announcement/anno
+
+sudo make -j -C $MIKU_PATH
+
```

Comment:
Add more about starting web server and judge server.
![image](https://user-images.githubusercontent.com/15229692/122893286-c6d02c80-d378-11eb-880a-b1425e7e364c.png)